shared/common-adapters/plain-input.js.flow

```diff
@@ -0,0 +1,86 @@
+// @flow
+import {Component} from 'react'
+import {type StylesCrossPlatform} from '../styles'
+import type {TextType} from './text'
+
+export type KeyboardType =
+  | 'default'
+  | 'email-address'
+  | 'numeric'
+  | 'phone-pad'
+  // iOS only
+  | 'ascii-capable'
+  | 'numbers-and-punctuation'
+  | 'url'
+  | 'number-pad'
+  | 'name-phone-pad'
+  | 'decimal-pad'
+  | 'twitter'
+  | 'web-search'
+  // Android Only
+  | 'visible-password'
+
+export type Props = {
+  autoFocus?: boolean,
+  className?: string,
+  disabled?: boolean,
+  // Resize in a flexbox-like fashion
+  flexable?: boolean,
+  maxLength?: number,
+  multiline?: boolean,
+  onBlur?: () => void,
+  onClick?: (event: Event) => void,
+  onChangeText?: (text: string) => void,
+  onFocus?: () => void,
+  placeholder?: string,
+  rowsMin?: number,
+  rowsMax?: number,
+  style?: StylesCrossPlatform,
+  textType?: TextType,
+  type?: 'password' | 'text' | 'number',
+
+  // NOTE: Setting this prop will make the input controlled, which
+  // can impair performance badly on some platforms
+  value?: ?string,
```

Comment:
It looks like the uncontrolled flag (and corresponding cases) aren't in this plain input, but then the code implicitly assumes that `state.value` is the source of truth, i.e. that the input is controlled.

I think we want to make uncontrolled inputs the default, or even to make it impossible to have a controlled input...